File: tournaments/zFunctionNaive/zFunctionNaive.js
function zFunctionNaive(s) {
    const n = s.length;
    const result = Array(n).fill(0);

    for (let i = 0; n > i; ++i) {
        for (let j = i; n > j; ++j) {
            if (s[j] === s[result[i]]) {
                ++result[i];
            } else {
                break;
            }
        }
    }

    return result;
}

File: tournaments/zFunctionNaive/zFunctionNaive.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub zFunctionNaive {
    my ($s) = @_;

    my @store = split('', $s);
    my $n = @store;
    my @result = (0)x$n;

    for (my $i = 0; $n > $i; ++$i) {
        for (my $j = $i; $n > $j; ++$j) {
            if ($store[$j] eq $store[$result[$i]]) {
                ++$result[$i];
            } else {
                last;
            }
        }
    }

    return \@result;
}
